### ISSUE TITLE
Pre_command/Post_command

### ISSUE THREAD

**Issue by tvanderhey** [controlm/ctm-python-client#57]
In the Job class, there is a pre_command and post_command attribute. I created a JobDatabase(Job) and JobCommand(Job) and added 'sleep 5m' to the pre_command for each job. The Command job slept for 5 minutes before running; however, the Database job did not sleep. Can the pre/post_commands be used with Database Jobs and if so, how can it be used?

**Comment by jfink5** [comment]
The usage of pre and post command should be consistent across all job types.  Did you have a look at the output when run?  If there was an error executing a command, then it should show up in there.  What platform did you use for the database job?  If windows, then know that windows shell does not have a "sleep" command.  But you may use Control-M Agent's _sleep.exe

**Comment by tvanderhey** [comment]
Both jobs are running in a linux environment. I created the jobs using a Python script and validated %%PRECMD was set for both jobs. However, when I ran the jobs only the OS job slept. There were no errors in the job output. Also, we have the web version of Control-M and when I view the jobs in the Planning tab, the OS job has a field for Pre_execution command, however, the Database job does not have a Pre_execution command field on the screen. I also tried _sleep.exe 300 for both jobs and neither one slept. I checked for errors and there were none.

SELECT name, value, application, group_name, job_name, appl_type
FROM emuser.def_ver_setvar x 
LEFT OUTER JOIN emuser.def_job y ON x.job_id = y.job_id AND x.table_id = y.table_id
WHERE name = '%%PRECMD' and x.is_current_version = 'Y';

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/LOCAL_~2.CT/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/LOCAL_~2.CT/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


NAME | VALUE | APPLICATION | GROUP_NAME | JOB_NAME | APPL_TYPE
-- | -- | -- | -- | -- | --
%%PRECMD | sleep 300 | TEST_PRE_POST_COMMANDS | TEST_PRE_POST_COMMANDS | ADE_REGISTRY.TJV_CONTROL_M_TEST_PKG.MAIN | DATABASE
%%PRECMD | sleep 300 | TEST_PRE_POST_COMMANDS | TEST_PRE_POST_COMMANDS | TEST_PRE_COMMAND | OS



</body>

</html>

**Comment by jfink5** [comment]
You are right.  It looks like some job types actually know how to use it.  So while its valid in the database model, the job type doesn't do anything with it.

**Comment by tvanderhey** [comment]
Thanks for checking. I was hoping I was doing something wrong, and it worked for all job types.
